packages/vite/src/node/utils.ts

```diff
@@ -757,10 +780,17 @@ export function resolveHostname(
   }
 
   // Set host name to localhost when possible
-  const name =
-    host === undefined || wildcardHosts.has(host) ? 'localhost' : host
+  let name = host === undefined || wildcardHosts.has(host) ? 'localhost' : host
+
+  if (host === 'localhost') {
+    // See #8647 for more details.
+    const localhostAddr = await getLocalhostAddressIfDiffersFromDNS()
+    if (localhostAddr) {
+      name = localhostAddr
+    }
+  }
 
-  return { host, name }
+  return { host, name, implicit: optionsHost === undefined }
```

Comment:
Should we have `implicit: optionsHost === undefined || optionsHost === false` like for line 772?

Tiny nit: What do you think if we rename `implicit` as `isLocal`, `local`, or inverse `public`? `implicit` sounds a bit vague to me unless it's a network term that I'm not familiar of 😅

Reply:
I added this property to differentiate whether the user set any value to `server.host`.
If a user set a value to `server.host`, I thought the message below is not needed.

> Network: Use **--host** to expose

If we should output this message even if the user set `localhost` to `server.host` explicitly, renaming the name and changing the value makes sense to me.

Reply:
I think that's a fair assumption, we could leave it as it then 👍